tools/Customization/DevHome.Customization/Views/AddRepositoriesView.xaml.cs

```diff
@@ -37,16 +38,46 @@ public void RemoveFolderButton_Click(object sender, RoutedEventArgs e)
         }
     }
 
-    public void AssignSourceControlProviderButton_Click(object sender, RoutedEventArgs e)
+    public async void AssignSourceControlProviderButton_Click(object sender, RoutedEventArgs e)
     {
         // Extract relevant data from view and give to view model for assign
         MenuFlyoutItem menuItem = (MenuFlyoutItem)sender;
         if (menuItem.DataContext is RepositoryInformation repoInfo)
         {
-            ViewModel.AssignSourceControlProviderToRepository(menuItem.Text, repoInfo.RepositoryRootPath);
+            var taskResult = await ViewModel.AssignSourceControlProviderToRepository(menuItem.Text, repoInfo.RepositoryRootPath);
+            if (taskResult?.Result != Helpers.ResultType.Success)
+            {
+                _log.Error("Error occurred while assigning source control provider: ", taskResult?.Error, taskResult?.Exception, taskResult?.DiagnosticText, taskResult?.DisplayMessage);
+                ShowErrorContentDialog(taskResult!, this.XamlRoot);
```

Comment:
I don't see any code resetting the Flyout state back to "Unassigned". I see that `AssignSourceControlProviderToRepository` refreshes the list on success, but are we going to leave the flyout/button in a misleading state on failure?

Reply:
In the code, errors occurring inside `AssignSourceControlProviderToRepository` cause an early return from the function (when validation fails) before the source control extension for the repository path is modified in the data store. Because of this the flyout, which binds to information relying on the data store as source of truth, still remains in the expected state i.e. unassigned and does not need a refresh.